File: src/junghyun/discord/ui/languages/LanguageENG.java
package junghyun.discord.ui.languages;

public class LanguageENG implements LanguageInterface {

    @Override
    public String[] TARGET_REGION() {
        return new String[]{};
    }

    @Override
    public String LANGUAGE_CODE() {
        return "ENG";
    }

    @Override
    public String LANGUAGE_NAME() {
        return "English:flag_gb:";
    }
    @Override
    public String LANGUAGE_DESCRIPTION() {
        return "Please use the `~lang` `ENG` command.";
    }

    @Override
    public String HELP_INFO() {
        return "GomokuBot / Help";
    }
    @Override
    public String HELP_DESCRIPTION() {
        return "GomokuBot is an open-source artificial intelligence Discord Bot that provides Gomoku(Omok) feature in Discord. The collected data is used for training reinforcement learning models.";
    }
    @Override
    public String HELP_DEV() {
        return "Developer";
    }
    @Override
    public String HELP_GIT() {
        return "Git Repository";
    }
    @Override
    public String HELP_VERSION() {
        return "Version";
    }
    @Override
    public String HELP_SUPPORT() {
        return "Support Channel";
    }
    @Override
    public String HELP_INVITE_LINK() {
        return "Invite Link";
    }

    @Override
    public String HELP_CMD_INFO() {
        return "GomokuBot / Command";
    }
    @Override
    public String HELP_CMD_HELP() {
        return "`~help` Get help";
    }
    @Override
    public String HELP_CMD_RANK() {
        return "`~rank` Show the ranking from 1st to 10th";
    }
    @Override
    public String HELP_CMD_LANG(String langList) {
        return "`~lang` " + langList + " Change the language setting used on this server. Ex) `~lang` `ENG`";
    }
    @Override
    public String HELP_CMD_SKIN() {
        return "`~skin` `A` `B` `C` Change the Gomoku-canvas style setting used on this server. Ex) `~skin` `A`";
    }
    @Override
    public String HELP_CMD_PVE() {
        return "`~start` Start the game with A.I.";
    }
    @Override
    public String HELP_CMD_PVP() {
        return "`~start` `@mention` Start the game with the mentioned player. Ex) `~start` `@player`";
    }
    @Override
    public String HELP_CMD_RESIGN() {
        return "`~resign` Surrender the current game.";
    }

    @Override
    public String SKIN_INFO() {
        return "GomokuBot / Style";
    }
    @Override
    public String SKIN_DESCRIPTION() {
        return "The default Gomoku canvas (Style A) may not display properly. Choose one of the three styles available and set the style to use on this server.";
    }
    @Override
    public String SKIN_CMD_INFO(String style) {
        return "Enter ``~skin`` ``" + style + "`` to use this style.";
    }
    @Override
    public String SKIN_CHANGE_ERROR() {
        return "There is an error in the style specification.";
    }
    @Override
    public String SKIN_CHANGE_SUCCESS(String style) {
        return "Style setting has been change to ``" + style + "`` !";
    }

    @Override
    public String RANK_INFO() {
        return "GomokuBot / Ranking";
    }
    @Override
    public String RANK_DESCRIPTION() {
        return "Ranked 1st to 10th.";
    }
    @Override
    public String RANK_WIN() {
        return "Victory";
    }
    @Override
    public String RANK_LOSE() {
        return "Defeat";
    }

    @Override
    public String LANG_CHANGE_ERROR() {
        return "There is an error in the language specification.";
    }
    @Override
    public String LANG_SUCCESS() {
        return "Language setting has been changed to English:flag_gb:!";
    }

    @Override
    public String GAME_NOT_FOUND(String nameTag) {
        return nameTag + ", could not find any games in progress. Please start the game with `~start` command!";
    }
    @Override
    public String GAME_CREATE_FAIL(String nameTag) {
        return nameTag + ", Game creation failed. Please finish the game in progress. :thinking:";
    }
    @Override
    public String GAME_SYNTAX_FAIL(String nameTag) {
        return nameTag + ", that's invalid command. Please write in the format of . `~s` `alphabet` `number` :thinking:";
    }
    @Override
    public String GAME_ALREADY_IN(String nameTag) {
        return nameTag + ", there is already a stone. :thinking:";
    }

    @Override
    public String GAME_CREATE_INFO(String playerName, String targetName, String fAttack) {
        return "The match between`" + playerName + "` vs `" + targetName + "` has begun! Attack first is `" + fAttack + "`.";
    }
    @Override
    public String GAME_CMD_INFO() {
        return "Please place the Stone by `~s` `alphabet` `number` format. Ex) `~s` `h` `8`";
    }

    @Override
    public String GAME_NEXT_TURN(String curName, String prvName, String lastPos) {
        return "`" + curName + "`, please place the next Stone. `" + prvName + "` was placed on " + lastPos;
    }

    @Override
    public String GAME_PVP_TURN(String turnName) {
        return "It's `"+ turnName + "`s turn now. Please wait for `"+ turnName +"`'s next stone. :thinking:";
    }
    @Override
    public String GAME_PVP_WIN(String winName, String loseName, String lastPos) {
        return "`" + winName + "` wins by `" + loseName + "` placing Stone on " + lastPos + "!";
    }
    @Override
    public String GAME_PVP_RESIGN(String winName, String loseName) {
        return "`" + winName + "` wins by `" + loseName + "` declaring surrender!";
    }

    @Override
    public String GAME_PVP_INFO(String winName, String loseName, int winCount, int loseCount) {
        return "`" + winName + "` vs `" + loseName + "` have been updated to `" + winCount + " : " + loseCount + "`.";
    }

    @Override
    public String GAME_PVE_WIN(String lastPos) {
        return "You beat AI by placing Stone on " + lastPos + ". Congratulations! :tada:";
    }
    @Override
    public String GAME_PVE_LOSE(String lastPos) {
        return "You have been defeated by AI placing on " + lastPos + ".";
    }
    @Override
    public String GAME_PVE_RESIGN() {
        return "You have been defeated by declaring surrender.";
    }

    @Override
    public String GAME_PVE_INFO(String playerName, int winCount, int loseCount, int rank) {
        return "Your entire AI has been updated to `" + winCount + " : " + loseCount + "`. " + playerName + " is currently ranked " + rank + " above.";
    }

    @Override
    public String GAME_FULL() {
        return "There was no more space for the stones, so it was a draw.";
    }

    @Override
    public String GAME_ARCHIVED(String messageLink) {
        return ":tada: You showed a great game! The game record has been shared to the official channel. - Click link to check it out.\n"+messageLink;
    }

    @Override
    public String BOARD_INP() {
        return "in Processing";
    }
    @Override
    public String BOARD_FINISH() {
        return "Finished";
    }
    @Override
    public String BOARD_TURNS() {
        return "Turn Progress";
    }
    @Override
    public String BOARD_TURN() {
        return "Turns";
    }
    @Override
    public String BOARD_LOCATION() {
        return "Latest Location";
    }

}

File: src/main/kotlin/i18n/LanguageENG.kt
package i18n

open class LanguageENG : LanguageInterface {

    override fun targetRegion(): Array<String> = arrayOf()

    override fun languageCode(): String = "ENG"

    override fun languageName(): String = "English:flag_gb:"
    override fun languageDescription(): String = "Please use the `~lang` `ENG` command."

    override fun helpInfo(): String = "GomokuBot / Help"
    override fun helpDescription(): String =
        "GomokuBot is an open-source artificial intelligence Discord Bot that provides Gomoku(Omok) feature in Discord. " +
                "The collected data is used for training reinforcement learning models."
    override fun helpDev(): String = "Developer"
    override fun helpRepo(): String = "Git Repository"
    override fun helpVersion(): String = "Version"
    override fun helpSupport(): String = "Support Channel"
    override fun helpInvite(): String = "Invite Link"

    override fun helpCmdInfo(): String = "GomokuBot / Command"
    override fun helpCmdHelp(): String = "`~help` Get help"
    override fun helpCmdRank(): String = "`~rank` Show ranking from 1st to 10th"
    override fun helpCmdLang(langList: String): String =
        "`~lang` $langList Change the language setting used on this server. Ex) `~lang` `ENG`"
    override fun helpCmdSkin(): String =
        "`~skin` `A` `B` `C` Change the Gomoku-canvas style setting used on this server. Ex) `~skin` `A`"
    override fun helpCmdPVE(): String = "`~start` Start the game with A.I."
    override fun helpCmdPVP(): String = "`~start` `@mention` Start the game with the mentioned player. Ex) `~start` `@player`"
    override fun helpCmdResign(): String = "`~resign` Surrender the current game."

    override fun skinInfo(): String = "GomokuBot / Style"
    override fun skinDescription(): String =
        "The default Gomoku canvas (Style A) may not display properly. " +
                "Choose one of the three styles available and set the style to use on this server."
    override fun skinCmdInfo(style: String): String = "Enter ``~skin`` ``$style`` to use this style."
    override fun skinUpdateError(): String = "There is an error in the style specification."
    override fun skinUpdateSuccess(style: String): String = "Style setting has been change to ``$style`` !"

    override fun rankInfo(): String = "GomokuBot / Ranking"
    override fun rankDescription(): String = "Ranked 1st to 10th."
    override fun rankWin(): String = "Victory"
    override fun rankLose(): String = "Defeat"

    override fun langUpdateError(): String = "There is an error in the language specification."
    override fun langUpdateSuccess(): String = "Language setting has been changed to English:flag_gb:!"

    override fun gameNotFound(nameTag: String): String =
        "$nameTag, could not find any games in progress. Please start the game with `~start` command!"
    override fun gameCreationFail(nameTag: String): String =
        "$nameTag, Game creation failed. Please finish the game in progress. :thinking:"
    override fun gameSyntaxFail(nameTag: String): String =
        "$nameTag, that's invalid command. Please write in the format of . `~s` `alphabet` `number` :thinking:"
    override fun gameAlreadyInProgress(nameTag: String): String = "$nameTag, there is already a stone. :thinking:"

    override fun gameCreateInfo(playerName: String, targetName: String, fAttack: String): String =
        "The match between`$playerName` vs `$targetName` has begun! Attack first is `$fAttack`."
    override fun gameCmdInfo(): String = "Please place the Stone by `~s` `alphabet` `number` format. Ex) `~s` `h` `8`"

    override fun gameNextTurn(curName: String, prvName: String, lastPos: String): String =
        "`$curName`, please place the next Stone. `$prvName` was placed on $lastPos"

    override fun gamePVPNextTurn(turnName: String): String =
        "It's `$turnName`s turn now. Please wait for `$turnName`'s next stone. :thinking:"
    override fun gamePVPWin(winName: String, loseName: String, lastPos: String): String =
        "`$winName` wins by `$loseName` placing Stone on $lastPos!"
    override fun gamePVPResign(winName: String, loseName: String): String =
        "`$winName` wins by `$loseName` declaring surrender!"

    override fun gamePVPInfo(winName: String, loseName: String, winCount: Int, loseCount: Int): String =
        "`$winName` vs `$loseName` have been updated to `$winCount : $loseCount`."

    override fun gamePVEWin(lastPos: String): String =
        "You beat A.I. by placing Stone on $lastPos. Congratulations! :tada:"
    override fun gamePVELose(lastPos: String): String = "You have been defeated by A.I. placing on $lastPos."
    override fun gamePVEResign(): String = "You have been defeated by declaring surrender."

    override fun gamePVEInfo(playerName: String, winCount: Int, loseCount: Int, rank: Int): String =
        "Your entire A.I. has been updated to `$winCount : $loseCount`. $playerName is currently ranked $rank above."

    override fun gameFull(): String = "There was no more space for the stones, so it was a draw."

    override fun boardInProgress(): String = "in Processing"
    override fun boardFinish(): String = "Finished"
    override fun boardProgress(): String = "Turn Progress"
    override fun boardTurns(): String = "Turns"
    override fun boardLocation(): String = "Latest Location"

}
